File: katas/6 kyu/count-the-smiley-faces/solution.py
# count-the-smiley-faces
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
import itertools
def count_smileys(arr):
  c = 0
  eye = [':' , ';']
  nose = ['' , '-' , '~']
  mouth = [')' , 'D']
  allowlist = ["".join(x) for x in itertools.product(eye,nose,mouth)]
  for x in arr:
    if x in allowlist:
      c += 1
  return c
# ------------------------------ CLEVER SOLUTION ----------------------------- #
from re import findall
def count_smileys(arr):
  return len(list(findall(r"[:;][-~]?[)D]", " ".join(arr))))
# ----------------------------------- TEST ----------------------------------- #
test.describe("Basic tests")
test.assert_equals(count_smileys([]), 0)
test.assert_equals(count_smileys([':D',':~)',';~D',':)']), 4)
test.assert_equals(count_smileys([':)',':(',':D',':O',':;']), 2)
test.assert_equals(count_smileys([';]', ':[', ';*', ':$', ';-D']), 1)

File: katas/6 kyu/count-the-smiley-faces/solution.php
<!-- count-the-smiley-faces -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function count_smileys($arr): int {
  return count(preg_grep("/[:;][-~]?[)D]/" , $arr));
}
/* ---------------------------------- TEST ---------------------------------- */
class MyTestCases extends TestCase {
  public function testBasics() {
    $this->assertEquals(0, count_smileys([]));
    $this->assertEquals(4, count_smileys([':D',':~)',';~D',':)']));
    $this->assertEquals(2, count_smileys([':)',':(',':D',':O',':;']));
    $this->assertEquals(1, count_smileys([';]', ':[', ';*', ':$', ';-D']));
  }
}